policy.go

```diff
@@ -1258,33 +840,51 @@ func executePolicySession(tpm *tpm2.TPMContext, policySession tpm2.SessionContex
 //
 // On success, subsequent calls to SealedKeyObject.UnsealFromTPM will fail with a ErrSealedKeyAccessLocked error until the next TPM
 // restart or TPM reset.
-func LockAccessToSealedKeys(tpm *TPMConnection) error {
+//
+// TODO: Update this documentation and maybe rename the function, as it doesn't make much sense now.
+func LockAccessToSealedKeys(tpm *TPMConnection, pcrs []int) error {
 	session := tpm.HmacSession()
 
-	handles, err := tpm.GetCapabilityHandles(lockNVHandle1, 1, session.IncludeAttrs(tpm2.AttrAudit))
+	// The fence is a hash of uint32(0), which is the same as EV_SEPARATOR (which can be uint32(0) or uint32(-1))
+	fence := make([]byte, 4)
+
+	// Insert PCR fence
+	for _, pcr := range pcrs {
+		seq, err := tpm.HashSequenceStart(nil, tpm2.HashAlgorithmNull)
+		if err != nil {
+			return xerrors.Errorf("cannot being hash sequence: %w", err)
+		}
+		if _, err := tpm.EventSequenceExecute(tpm.PCRHandleContext(pcr), seq, fence, session, nil); err != nil {
+			return xerrors.Errorf("cannot execute hash sequence: %w", err)
+		}
+	}
+
+	// Handle legacy locking for v0 key files
```

Comment:
do we strictly need to keep this bit? is it because the old v0 keys likely weren't using the binary digests? the scrambling of the PCRs will affect all keys new and old right?

Reply:
Given that we know all of the keys created by this package use the same PCR profile right now, it probably is fine to drop this.